components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineView.kt

```diff
@@ -243,19 +250,20 @@ class SystemEngineView @JvmOverloads constructor(
                                 view.post { view.loadUrl(url) }
                                 super.shouldInterceptRequest(view, request)
                             }
+                            is InterceptionResponse.AppIntent -> {
+                                appIntent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
+                                try {
+                                    context.startActivity(appIntent)
+                                } catch (e: ActivityNotFoundException) {
+                                    Logger.error("failed to start third party app activity", e)
+                                }
+                                super.shouldInterceptRequest(view, request)
+                            }
                         }
                     }
                 }
             }
 
-            if (request.isForMainFrame) {
```

Comment:
I'm not sure we should be removing the `isForMainFrame` check here.

Reply:
I'll look into this more.

Reply:
Agreed that I shouldn't ignore observers for `SystemEngineView`.  Updated to match the design of `GeckoEngineSession` so its more consistent.  Now `request.isForMainFrame` is checked when handling the response from the interceptor.